### ISSUE TITLE
Train a tiny model

### ISSUE THREAD

**Issue by yipingji** [allenai/OLMo#814]
### 🐛 Describe the bug

Hi,

I tried to train tiny models and I replaced data path starting with "s3://ai2-llm/" by "https://olmo-data.org/" in the config files. 

Then I ran "torchrun --nproc_per_node=2  scripts/train.py /raid/work/ji016/OLMo/configs/tiny/OLMo-60M.yaml --save_overwrite"

![Image](https://github.com/user-attachments/assets/92741eb5-632f-4d86-b539-2ecd03733724)

### Versions

Python 3.10.16                                                                                                                                               
-e git+https://github.com/allenai/OLMo.git@a87c459d038c049045b09a05c4987fdddb01393e#egg=ai2_olmo                                                             
ai2-olmo-core==2.0.1                                                                                                                                         
ai2-olmo-eval==0.7.0                                                                                                                                         
aiohappyeyeballs==2.6.1                                                                                                                                      
aiohttp==3.11.14                                                                                                                                             
aiosignal==1.3.2                                                                                                                                             
annotated-types==0.7.0                                                                                                                                       
antlr4-python3-runtime==4.9.3                                                                                                                                
async-timeout==5.0.1                                                                                                                                         
attrs==25.3.0                                                                                                                                                
boto3==1.37.19                                                                                                                                               
botocore==1.37.19                                                                                                                                            
cached_path==1.7.1                                                                                                                                           
cachetools==5.5.2                                                                                                                                            
certifi==2025.1.31                                                                                                                                           
charset-normalizer==3.4.1                                                                                                                                    
click==8.1.8                                                                                                                                                 
datasets==3.4.1                                                                                                                                              
dill==0.3.8                                                                                                                                                  
docker-pycreds==0.4.0                                                                                                                                        
filelock==3.18.0                                                                                                                                             
frozenlist==1.5.0                                                                                                                                            
fsspec==2024.12.0                                                                                                                                            
gitdb==4.0.12                                                                                                                                                
GitPython==3.1.44                                                                                                                                            
google-api-core==2.24.2                                                                                                                                      
google-auth==2.38.0                                                                                                                                          
google-cloud-core==2.4.3                                                                                                                                     
google-cloud-storage==2.19.0                                                                                                                                 
google-crc32c==1.7.1rc1                                                                                                                                      
google-resumable-media==2.7.2                                                                                                                                
googleapis-common-protos==1.69.2                                                                                                                             
huggingface-hub==0.27.1                                                                                                                                      
idna==3.10                                                                                                                                                   
importlib_resources==6.5.2                                                                                                                                   
Jinja2==3.1.6                                                                                                                                                
jmespath==1.0.1                                                                                                                                              
joblib==1.4.2                                                                                                                                                
lightning-utilities==0.14.2                                                                                                                                  
markdown-it-py==3.0.0                                                                                                                                        
MarkupSafe==3.0.2                                                                                                                                            
mdurl==0.1.2                                                                                                                                                 
mpmath==1.3.0                                                                                                                                                
multidict==6.2.0                                                                                                                                             
multiprocess==0.70.16                                                                                                                                        
networkx==3.4.2                                                                                                                                              
numpy==1.26.4                                                                                                                                                
nvidia-cublas-cu12==12.4.5.8                                                                                                                                 
nvidia-cuda-cupti-cu12==12.4.127                                                                                                                             
nvidia-cuda-nvrtc-cu12==12.4.127                                                                                                                             
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-cusparselt-cu12==0.6.2
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
omegaconf==2.3.0
packaging==24.2
pandas==2.2.3
pillow==11.1.0
platformdirs==4.3.7
propcache==0.3.0
proto-plus==1.26.1
protobuf==5.29.4
psutil==7.0.0
pyarrow==19.0.1
pyasn1==0.6.1
pyasn1_modules==0.4.1
pydantic==2.10.6
pydantic_core==2.27.2
Pygments==2.19.1
python-dateutil==2.9.0.post0
pytz==2025.2
PyYAML==6.0.2
regex==2024.11.6
requests==2.32.3
rich==13.9.4
rsa==4.9
s3transfer==0.11.4
safetensors==0.5.3
scikit-learn==1.6.1
scipy==1.15.2
sentry-sdk==2.24.1
setproctitle==1.3.5
six==1.17.0
smmap==5.0.2
sympy==1.13.1
threadpoolctl==3.6.0
tokenizers==0.21.1
torch==2.6.0
torchaudio==2.6.0
torchmetrics==1.7.0
torchvision==0.21.0
tqdm==4.67.1
transformers==4.50.0
triton==3.2.0
typing_extensions==4.12.2
tzdata==2025.2
urllib3==2.3.0
wandb==0.19.8
xxhash==3.5.0
yarl==1.18.3

**Comment by ved1beta** [comment]
you replaced the S3 URLs with https://olmo-data.org/, but there's no indication that the OLMo codebase supports loading training data from HTTP URLs. The data loader is likely designed specifically for S3 or local file paths.
The repository structure doesn't provide a clear way for external users to access the training data. There is no public endpoint at olmo-data.org that mirrors the S3 bucket structure.
The configuration files in configs/tiny are designed to work with AI2's internal infrastructure and S3 data storage.

the repository includes test fixtures for training tiny models. Instead of trying to access the full training data, you should
Use the test configuration as a starting point. Copy from test_fixtures/train_tiny.yaml which uses a smaller sample dataset:
```data:
  paths:
    - "/tmp/c4-sample.npy"
  persistent_workers: false
  num_workers: 0
  prefetch_factor: null
tokenizer:
  identifier: "gpt2"  # Use GPT-2 tokenizer instead of custom one```

**Comment by ved1beta** [comment]
Hi @yipingji,

I understand you're trying to train a tiny OLMo model and encountering issues with data access. Let me explain what's happening and how to fix it:

Problem Explanation:
------------------
The error occurs because you're trying to replace S3 URLs (s3://ai2-llm/) with HTTPS URLs (https://olmo-data.org/). 
This won't work because:
1. The OLMo training infrastructure is specifically designed to work with either local files or S3 buckets
2. There isn't actually a public HTTPS endpoint at olmo-data.org that mirrors the S3 structure
3. The config files in configs/tiny/ are designed for AI2's internal infrastructure

Solutions (in order of simplicity):
---------------------------------
1. Use Test Setup (Easiest):
   - Use test_fixtures/train_tiny.yaml as your starting point
   - It uses a small sample dataset (/tmp/c4-sample.npy)
   - Uses the public GPT-2 tokenizer instead of OLMo's custom tokenizer
   - Example command:
     ```bash
     torchrun --nproc_per_node=2 scripts/train.py test_fixtures/train_tiny.yaml
     ```

2. Use Public Datasets:
   - OLMo's training data is available publicly on Hugging Face:
     * https://huggingface.co/datasets/allenai/olmo-mix-1124
     * https://huggingface.co/datasets/allenai/dolmino-mix-1124
   - You'll need to convert these to the expected .npy format

3. Get Official Access:
   - Join AI2's Discord: https://discord.gg/sZq3jTNVNG
   - Ask about proper access to training data
   - They can guide you on setting up proper S3 access or using alternative data sources

Modified Config Example:
----------------------
Here's how you could modify your config:
```yaml
run_name: OLMo-60M-test
# Keep your model architecture settings from OLMo-60M.yaml
model:
  d_model: 512
  n_heads: 8
  n_layers: 8
  # ... other model settings ...

# Replace data section with this:
data:
  paths:
    - "/tmp/c4-sample.npy"
  persistent_workers: false
  num_workers: 0
  prefetch_factor: null

# Use public tokenizer instead of custom one
tokenizer:
  identifier: "gpt2"
```

I recommend starting with solution #1 (test setup) to get things working locally first, then 
moving to the public datasets or getting official access once you're ready to train on full data.

Let me know if you need any clarification or run into other issues!

**Comment by aman-17** [comment]
Fixed in #828
